### PR TITLE
Make filenames flexible for Python assignments

### LINKED ISSUE #8: Flexible filenames
multi-checker expects assignment N in a file called `aufgabe_0N.py` if the filename is not specified in exercise.yaml.

We should be more flexible:

- don't care about lower and upper case
- also accept `aufgabe_N.py`, `aufgabeN.py`, `aufgabe0N.py`, `aufgabe-0N.py`, and `aufgabe-N.py`

### PR DESCRIPTION
**Pull request by hannesbraun**
Closes #8

### CODE REVIEW

File: multi-checker/script/utils.py [hunk 1]
```diff
@@ -53,15 +53,20 @@ def fixEncodingRecursively(startDir: str, ext: str):
...
         return cand
     for x in ls(sourceDir):
+        if ignoreCase and basename(x).lower() == p.lower() and isFile(x):
+            return x
         if isDir(x):
             cand = pjoin(x, p)
             if isFile(cand):
                 return cand
+            for y in ls(x):
+                if ignoreCase and basename(y).lower() == p.lower() and isFile(y):
```
Comment (skogsbaer) [inline multi-checker/script/utils.py]: Where is the logic where we descend into the directory tree until we reach a directory that contains more than one subdirectory? There seems to be some duplicated logic.

Comment (hannesbraun) [reply multi-checker/script/utils.py]: That's contained in the findSolutionDir method in the utils.py file. We can probably remove the block inside of the `if isDir(x):` block. What do you think?

Comment (skogsbaer) [reply multi-checker/script/utils.py]: Yes, I think this should work.  The right way to find out is to ensure that a test for this feature exists. If it still runs after removing the block, everything is fine ;-)

Comment (hannesbraun) [reply multi-checker/script/utils.py]: There is a test for this behavior (see `{pythonTestDir}/labortest_2/solution/`) and it fails after removing the block.


I'm not sure why we would want to support spreading solutions across multiple directories:
```
solution
├── aufgabe_01.py
└── subdir
    ├── aufgabe_01.py
    ├── aufgabe_02.py
    └── aufgabe_03.py
```

Comment (skogsbaer) [reply multi-checker/script/utils.py]: This generality is not needed, indeed. So remove it!

Comment (hannesbraun) [reply multi-checker/script/utils.py]: Done

**Pull request merged**